hscontrol/types/acl.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"google.golang.org/protobuf/types/known/structpb"
+	"gorm.io/datatypes"
+
+	v1 "github.com/juanfont/headscale/gen/go/headscale/v1"
+)
+
+var (
+	ErrACLPolicyNotFound         = errors.New("acl policy not found")
+	ErrInvalidACLPolicyFormat    = errors.New("invalid policy format")
+	ErrACLPolicyUpdateIsDisabled = errors.New("update is disabled for modes other than 'db'")
+)
+
+// ACL describes the data model for ACLs used to restrict access to resources.
+type ACL struct {
```

Comment:
There was a proposal of having the policy versioned. I do not think we should do that now, but I think we should think of if there is anything about the format that should be addressed to enable that in the future. I do not have any concerns per now, but want to hear what you think.

Reply:
> having the policy versioned. 

I agree. Having plus one line of a simple `version=2024-03-24` is not a big deal now, 
but can save us lots of trouble later.

_(In my 30y. of experience I've realized: 
 - it is better to use ISO-dates for versions. 

Saves lot's of time and prevents confusion. 
Also better to merge with release-notes / what changed.)_